py-polars/polars/functions/col.py

```diff
@@ -114,6 +121,8 @@ def _polars_dtype_match(tp: PolarsDataType) -> list[PolarsDataType]:
         return list(DATETIME_DTYPES)
     elif Duration.is_(tp):
         return list(DURATION_DTYPES)
+    elif Categorical.is_(tp):
+        return list(CATEGORICAL_DTYPES)
```

Comment:
This is needed such that selecting `pl.Categorical` in `pl.col` yields all categorical columns, regardless of ordering